src/actions/layers/Layers.ts

```diff
@@ -2,29 +2,65 @@ import {ILayerAction} from "./ILayerAction";
 import Action from "../Action";
 // TODO - BundleSize Warning - we include all the Sources code within Layers.
 import Source, {ImageSource} from "./Source";
+import {Position} from "../../params/position/Position";
+import Param from "../../parameters/Param";
 
 
 class Layer extends Action implements ILayerAction {
   source: ImageSource; // TODO this needs to accept other types of sources
-  constructor(transformable: ImageSource) {
+  position:Position;
+  constructor(transformable: ImageSource, position:Position) {
     super();
     this.source = transformable;
+    this.position = position;
+  }
+
+  /**
+   * Layers are built using three bits -> /Open/Transform/Close
+   * The opening of a layer
+   */
+  openLayer() {
+    return `l_${this.source.asset.publicID}`;
```

Comment:
This only supports regular image layers - when adding video/fetch/whatever this function will look differently I assume.

Reply:
Yes this is specific to images, videos will be added and changes will be done where needed.